### PR TITLE
Add more logging and small enhacement to an int test

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes - no work item, need more logging for investigation.

**Changes proposed in this request**
- log app config hash code, to be able to tell if requests belong to same app
- rename DefaultLogger to AppLogger
- enhance an integration test to use WithAuthority at request level

**Testing**

**Performance impact**

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs [hunk 1]
```diff
@@ -55,16 +55,16 @@ public void TestInitialize()
...
-            await RunOnBehalfOfTestAsync(aadUser1, true).ConfigureAwait(false);
-            await RunOnBehalfOfTestAsync(aadUser2, false).ConfigureAwait(false);
-            await RunOnBehalfOfTestAsync(adfsUser, true).ConfigureAwait(false);
-            await RunOnBehalfOfTestAsync(aadUser2, true).ConfigureAwait(false);
-            await RunOnBehalfOfTestAsync(aadUser2, false, true).ConfigureAwait(false);
+            //await RunOnBehalfOfTestAsync(aadUser1, true).ConfigureAwait(false);
+            //await RunOnBehalfOfTestAsync(aadUser2, false).ConfigureAwait(false);
+            //await RunOnBehalfOfTestAsync(adfsUser, true).ConfigureAwait(false);
+            //await RunOnBehalfOfTestAsync(aadUser2, true).ConfigureAwait(false);
+            //await RunOnBehalfOfTestAsync(aadUser2, false, true).ConfigureAwait(false);
```
Comment (neha-bhargava) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs:67]: Any reason for commenting these?

File: src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs [hunk 2]
```diff
@@ -338,23 +308,65 @@ protected Task ResolveAuthorityAsync()
...
+                authenticationRequestParameters.RequestContext.Logger.Error("The default token cache provided by MSAL is not designed to be performant when used in confidential client applications. Please use token cache serialization. See https://aka.ms/msal-net-cca-token-cache-serialization.");
+            }
+        }
+
         private void LogReturnedToken(AuthenticationResult result)
         {
             if (result.AccessToken != null && 
                 AuthenticationRequestParameters.RequestContext.Logger.IsLoggingEnabled(LogLevel.Info))
             {
+                int appHashCode = AuthenticationRequestParameters.AppConfig.GetHashCode();
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs:357]: - Maybe rename here and in log below to appConfigHashCode?
- In which scenario would this hashCode be useful?

Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs:357]: I think the hash code is useful to know the instance of confidential client being used to track how many CCA are created.

**Review by trwalke — APPROVED** [review]
I think this is a good idea.
LGTM

**Review by pmaytak — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs (hunk 1 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs:67]: Good catch, reverting.

File: src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs (hunk 2 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs:357]: Yes, just as Neha stated, it's used to differentiate between different CCA objects. It's to be able to tell if ppl are using 1 or more instances of CCA object, as it has cache implications.

**Pull request merged**